dist/lucca-ui.js

```diff
@@ -872,6 +873,437 @@
+
+		/****************/
+		/***** FIND *****/
+		/****************/
+
+		$scope.find = function (clue) {
+			getUsersAsync(clue).then(
+				function(users) {
+					var filteredUsers = users;
+					var count = filteredUsers.length;
+
+					// Used for UserPickerMultiple only
+					if (ctrl.isMultipleSelect) {
+						// Remove duplicates between results and selected users
+						_.each($scope.selected.users, function (selectedUser) {
+							filteredUsers = _.reject(filteredUsers, function (user) {
+								return (user.id === selectedUser.id);
+							});
+						});
+						// Number of results matching the query
+						count = filteredUsers.length;
+					}
+
+					// Used when a custom filtering function is given
+					if (ctrl.useCustomFilter) {
+						// Get new set of users
+						filteredUsers = _.filter(filteredUsers, function(user){ return $scope.customFilter(angular.copy(user)); });
+						count = filteredUsers.length;
+					}
+
+					if (hasFormerEmployees(filteredUsers)) {
+						handleFormerEmployees(filteredUsers);
+					}
+
+					if (hasHomonyms(filteredUsers)) {
+						tagHomonyms(filteredUsers);
+						handleHomonyms(filteredUsers);
+					}
+
+					if (hasPagination(filteredUsers)) {
+						if (asyncPagination) {
+							filteredUsers = addOverflowMessage(filteredUsers, "...");
+							// Call the api to get the right count and update the message to the user
+							handlePaginationAsync(clue).then(function(cnt){
+								count = cnt;
+								filteredUsers = updateOverflowMessage(filteredUsers, MAX_COUNT, count);
+							}, function(error) {
+								// HANDLE ERROR
+							});
+						}
+						else {
+							filteredUsers = addOverflowMessage(filteredUsers, count);
+						}
+					}
+					$scope.users = filteredUsers;
+					$scope.count = count;
+				}, 
+				function(error) {
+					// HANDLE ERROR
+				}
+			);
+		};
+
+		/*******************/
+		/***** TIMEOUT *****/
+		/*******************/
+
+		var reinitTimeout = function() {
+			// Cancel previous timeout
+			if (timeout.count) {
+				$timeout.cancel(timeout.count);
+			}
+		};
+
+		/*****************/
+		/***** USERS *****/
+		/*****************/
+
+		var getLimit = function() {
+			var limit = MAGIC_NUMBER_maxUsers;
+
+			if (ctrl.asyncPagination) {
+				limit = MAX_COUNT + 1;
+			}
+			return limit;
+		};
+
+		var getUsersAsync = function(input) {
+			var formerEmployees = "formerEmployees=" + ($scope.showFormerEmployees ? "true" : "false");
+			var limit = "&limit=" + getLimit();
+			var clue = "&clue=" + input;
+			var query = "/api/v3/users/find?" + (init ? "" : (clue + "&")) + formerEmployees + limit;
+			var deferred = $q.defer();
+
+			// Happen when the user starts typing a name, then waits enough to call the api and continues typing
+			// We do not want to treat the result of the previous request since they are now obsolete
+			if (promise) {
+				promise.then(function (response) {}); // do nothing with the results
+			}
+
+			reinitTimeout();
```

Comment:
`getUserAsync` shouldn't be the one calling `reinitTimeout` or canceling `promise`. It has to do with the async pagination and should not be referenced in the part in charge of fetching a list of user